sdk/testing/stepwise/environments/docker/environment.go

```diff
@@ -0,0 +1,851 @@
+	ServerKeyPEM      []byte
+	ServerKeyPEMFile  string
+	TLSConfig         *tls.Config
+	WorkDir           string
+	Cluster           *DockerCluster
+	container         *types.ContainerJSON
+	dockerAPI         *docker.Client
+}
+
+// CreateAPIClient creates and configures a Vault API client to communicate with
+// the running Vault Cluster for this DockerClusterNode
+func (n *DockerClusterNode) CreateAPIClient() (*api.Client, error) {
+	transport := cleanhttp.DefaultPooledTransport()
+	transport.TLSClientConfig = n.TLSConfig.Clone()
+	if err := http2.ConfigureTransport(transport); err != nil {
+		return nil, err
+	}
+	client := &http.Client{
+		Transport: transport,
+		CheckRedirect: func(*http.Request, []*http.Request) error {
+			// This can of course be overridden per-test by using its own client
+			return fmt.Errorf("redirects not allowed in these tests")
+		},
+	}
+	config := api.DefaultConfig()
+	if config.Error != nil {
+		return nil, config.Error
+	}
+	config.Address = fmt.Sprintf("https://127.0.0.1:%s", n.HostPort)
+	config.HttpClient = client
+	config.MaxRetries = 0
+	apiClient, err := api.NewClient(config)
+	if err != nil {
+		return nil, err
+	}
+	apiClient.SetToken(n.Cluster.RootToken())
+	return apiClient, nil
+}
+
+// Cleanup kills the container of the node
+func (n *DockerClusterNode) Cleanup() error {
+	return n.dockerAPI.ContainerKill(context.Background(), n.container.ID, "KILL")
+}
+
+func (n *DockerClusterNode) Start(cli *docker.Client, caDir, netName string, netCIDR *DockerClusterNode, pluginBinPath string) error {
+	n.dockerAPI = cli
+
+	err := n.setupCert()
+	if err != nil {
+		return err
+	}
+
+	vaultCfg := map[string]interface{}{
+		"listener": map[string]interface{}{
+			"tcp": map[string]interface{}{
+				"address":       fmt.Sprintf("%s:%d", "0.0.0.0", 8200),
+				"tls_cert_file": "/vault/config/cert.pem",
+				"tls_key_file":  "/vault/config/key.pem",
+				"telemetry": map[string]interface{}{
+					"unauthenticated_metrics_access": true,
+				},
+			},
+		},
+		"telemetry": map[string]interface{}{
+			"disable_hostname": true,
+		},
+		"storage": map[string]interface{}{
+			"raft": map[string]interface{}{
+				"path":    "/vault/file",
+				"node_id": n.NodeID,
+			},
+		},
+		"cluster_name":         netName,
+		"log_level":            "TRACE",
+		"raw_storage_endpoint": true,
+		"plugin_directory":     "/vault/config",
+		// disable_mlock is required for working in the Docker environment with
+		// custom plugins
+		"disable_mlock": true,
+		// These are being provided by docker-entrypoint now, since we don't know
+		// the address before the container starts.
+		//"api_addr": fmt.Sprintf("https://%s:%d", n.Address.IP, n.Address.Port),
+		//"cluster_addr": fmt.Sprintf("https://%s:%d", n.Address.IP, n.Address.Port+1),
+	}
+	cfgJSON, err := json.Marshal(vaultCfg)
+	if err != nil {
+		return err
+	}
+
+	err = ioutil.WriteFile(filepath.Join(n.WorkDir, "local.json"), cfgJSON, 0644)
+	if err != nil {
+		return err
+	}
```

Comment:
Can this be in a different function?

Reply:
Deferring and tracking this internally, to follow up in the coming weeks